index.js

```diff
@@ -1,20 +1,606 @@
+  while (answer !== "exit") {
+    //the user has said yes - the user begins in the first room and is shown the room description
+    if (answer.toLowerCase().trim().includes("yes")) {
+      //sets the currentRoom as the greatHall.  This is stored in the player object
+      player.currentRoom = greatHall;
+      greatHall.enter();
+    }
+
+    //prompts user for input
+    answer = await ask(">_");
+
+    //sanitize answer
+    answer = answer.toLowerCase().trim();
+
+    //check if the player input is "exit" - if so, end the game
+    if (answer === "exit") {
+      console.log(
+        '"Hogwarts will always be there to welcome you home"\n\t-J.K.Rowling'
+      );
+      console.log("Goodbye");
+      process.exit();
+    }
+
+    //breaks the answer into action and item (noun)
+    let answerArr = splitAnswer(answer);
+
+    //declares variables to hold the answerAction (action) and answerItem (noun)
+    let answerAction = answerArr[0];
+    let answerItem = answerArr[1];
+
+    //only if both answerAction and answerItem are defined will the the switch statement be triggered.  Otherwise the user input is not valid
+    if (answerAction && answerItem) {
+      //switch statement looks at answerAction and triggers the appropriate case based on input
+      switch (answerAction) {
+        case "move":
+          //vet the direction against this array before entering move case
+          // let directionArray = ["north", "east", "south", "west"];
+          player.currentRoom.move(answerItem);
+          break;
+        //unlock is triggered to open the roomOfRequirement, we want to change the current room to whichever room belongs to the desired direction key, then use our move function to move into the room
+        case "unlock":
+          currentRoom = player.currentRoom;
+          //since answer item is direction, currentRoom[answerItem] is identify what room you're moving too: in this case it's the roomOfRequirement
+          let newRoom = roomKey[currentRoom[answerItem]];
+          //keeps new room unlocked
+          newRoom.isUnlocked = true;
+          player.currentRoom.move(answerItem);
+          break;
+        //if answerAction is look - trigger lookAround method
+        case "look":
+          player.currentRoom.lookAround();
+          break;
+        //if answerAction is drop - trigger drop method
+        case "drop":
+          //calls the drop method on the item class
+          console.log(itemKey[answerItem].drop());
+          //adds the dropped item to the current room's inventory
+          player.currentRoom.itemsInRoom.push(itemKey[answerItem].name);
+          break;
+        //if answerAction is check - trigger checkInventory
+        case "check":
+          checkPlayerInventory();
+          break;
+        //if answerAction is examine - trigger examine method
+        case "examine":
+          console.log(itemKey[answerItem].examine());
+          break;
```

Comment:
I hit this switch case at one point in a playthrough and it caused the game to break. I think i tried to examine a specific item but I couldn't replicate the bug after it happened the first time